### PR TITLE
FreeBSD support in .travis-ocaml.sh

### PR DESCRIPTION
**Pull request by cyrilst**
Hello, here is a PR for FreeBSD support in .travis-ocaml.sh

I tried it successfully with https://github.com/geneweb/geneweb (OCAML_VERSION=4.05)

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
I looked around a bit and couldn't find any documentation saying that Travis now supports FreeBSD. Could you provide the appropriate documentation and add FreeBSD to the list of tested distribution in the `travis.yml` file?

**Comment by cyrilst** [comment]
Indeed, travis documentation is not yet up to date (discussion here : https://github.com/travis-ci/travis-build/pull/1780 )

I've just lauched a travis run with freebsd, we'll see if it's ok…

**Comment by cyrilst** [comment]
Looks like all tests passed without problems (one mac os test had to be restarted, but it's unrelated) : https://travis-ci.org/github/cyrilst/ocaml-ci-scripts

File: .travis-ocaml.sh [hunk 1]
```diff
@@ -99,6 +99,8 @@ install_ocaml () {
 
 install_opam2 () {
     case $TRAVIS_OS_NAME in
+        freebsd)
+            sudo pkg install -qy ocaml-opam ;;
```
Comment (kit-ty-kate) [inline .travis-ocaml.sh:103]: would it be ok to find some ways of having something similar to what happens on linux or macos?

**Comment by XVilka** [comment]
I am not sure about FreeBSD in Travis, but we use [SourceHut](https://sr.ht) for FreeBSD and OpenBSD (NetBSD support WIP also), see https://github.com/ocaml/ocaml-ci-scripts/issues/307

File: .travis-ocaml.sh (hunk 1 above)
Comment (cyrilst) [reply .travis-ocaml.sh:103]: Yes, it's pretty easy, we just need to add freebsd in the releases here : https://github.com/ocaml/opam/releases
I have no idea of how to do this…

Comment (kit-ty-kate) [reply .travis-ocaml.sh:108]: let's just give up on this for the moment and take the opam from the FreeBSD distribution. I forbid users from setting OPAM_VERSION or LOCAL_INSTALL. We can always allow them in the future.

**Comment by kit-ty-kate** [comment]
I'll let CI running and this PR should be good to go now.

cc @hannesm and @mato, this might interest you in the short term.

**Comment by cyrilst** [comment]
Awesome, thank you!

**Comment by kit-ty-kate** [comment]
ok let's try! Thanks!

**Comment by hannesm** [comment]
this is great, thanks for your contribution. It'd be even better if the support would be mentioned in the documentation (README-travis.md has a section `### Testing on different OS` which could be extended).

**Comment by cyrilst** [comment]
That's totally right: https://github.com/ocaml/ocaml-ci-scripts/pull/334

**Pull request merged**